app/src/ereferrals/java/org/eyeseetea/malariacare/data/remote/ElementController.java

```diff
@@ -0,0 +1,50 @@
+package org.eyeseetea.malariacare.data.remote;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.util.Log;
+
+import com.element.utils.ElementSDKManager;
+
+import org.eyeseetea.malariacare.domain.boundary.IElementController;
+
+public class ElementController implements IElementController, ElementSDKManager.ElementActivityListener, ElementSDKManager.EnrollListener {
+
+    final private String TAG = "ElementController";
+
+    public interface Callback {
+        void onSuccess(String uid);
+
+        void onError();
+    }
+
+    private Activity mActivity;
+    private Callback mCallback;
+
+    public ElementController(Callback callback, Activity activity){
+        mCallback = callback;
+        mActivity = activity;
+    }
+
+    @Override
+    public void sendVoucher(String voucherUId) {
```

Comment:
This is a bit confusing. Actually, it's not sending a voucher, but the UID to Element, for it to deal with it. I think the interface methods naming should be changing according to my previous comment, and do what it's intended to do. Maybe the interface could be prepared to receive the whole voucher (domain entity) and element controller just take what it needs (voucher id) and pass it to element? Something like a post-send voucher action, that in this case, for element, is just send the voucher id.